File: backend/IMIS/IMIS.Domain/IsoAuditor.cs
using Base.Primitives;


namespace IMIS.Domain
{
    public class IsoAuditor : Entity<int>
    {
        public required int? TeamId { get; set; }
        public Team? Team { get; set; }

        public required string? AuditorId { get; set; }
        public Auditor? IsoAuditors { get; set; }

        public required int? AuditPlanEntryId { get; set; }
        public AuditPlanEntry? AuditPlanEntry { get; set; }
    }
}

File: frontend/imis/lib/audit/audit_plan/models/iso_auditor.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:imis/team/models/team.dart'; // 👈 Check if this path is valid

part 'iso_auditor.g.dart';

@JsonSerializable(explicitToJson: true)
class IsoAuditor {
  @JsonKey(defaultValue: 0)
final int id;
@JsonKey(defaultValue: 0)
final int teamId;
final Team? team;
final int? auditorId;
@JsonKey(defaultValue: 0)
final int auditPlanEntryId;

IsoAuditor({
  this.id = 0,
  this.teamId = 0,
  this.team,
  this.auditorId,
  this.auditPlanEntryId = 0,
});

  factory IsoAuditor.fromJson(Map<String, dynamic> json) =>
      _$IsoAuditorFromJson(json);

  Map<String, dynamic> toJson() => _$IsoAuditorToJson(this);
}